core/src/text/fontContext.cpp

```diff
@@ -294,6 +294,7 @@ void FontContext::releaseFonts() {
     // those are 'weak' resources (would be automatically reloaded by alfons from
     // its URI or source callback.
     for (auto& font : m_font) {
+        if (!font) { continue; }
```

Comment:
We can do slightly better actually. If even a single font face is set, then all 3 sizes are set, so no need to call the for loop.
Another thing is to maintain a state which gets set in `loadFonts` and not call `releaseFonts` if this state is not set. Slightly cleaner and logical in my opinion.

Reply:
No performance concern here. Anything more abstracted is gonna start making assumptions about code elsewhere and I'd rather just keep it dumb.